services/docker-compose.yml

```diff
@@ -194,43 +194,77 @@ services:
+      RABBIT_PORT: ${RABBIT_PORT}
+      RABBIT_USER: ${RABBIT_USER}
+      RABBIT_PASSWORD: ${RABBIT_PASSWORD}
+      RABBIT_CHANNELS: ${RABBIT_CHANNELS}
+      REDIS_HOST: ${REDIS_HOST}
+      REDIS_PORT: ${REDIS_PORT}
+      POSTGRES_ENDPOINT: ${POSTGRES_ENDPOINT}
+      POSTGRES_USER: ${POSTGRES_USER}
+      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
+      POSTGRES_DB: ${POSTGRES_DB}
+      POSTGRES_HOST: ${POSTGRES_HOST}
+      POSTGRES_PORT: ${POSTGRES_PORT}
+      S3_ENDPOINT: ${S3_ENDPOINT}
+      S3_ACCESS_KEY: ${S3_ACCESS_KEY}
+      S3_SECRET_KEY: ${S3_SECRET_KEY}
+      S3_BUCKET_NAME: ${S3_BUCKET_NAME}
+      STORAGE_ENDPOINT: ${STORAGE_ENDPOINT}
+      REGISTRY_URL: ${REGISTRY_URL}
+      REGISTRY_PATH: ${REGISTRY_PATH}
+      REGISTRY_USER: ${REGISTRY_USER}
+      REGISTRY_PW: ${REGISTRY_PW}
+      SWARM_STACK_NAME: ${SWARM_STACK_NAME:-simcore}
+      SIDECAR_LOGLEVEL: ${LOG_LEVEL:-WARNING}
+      SIDECAR_HOST_HOSTNAME_PATH: ${SIDECAR_HOST_HOSTNAME_PATH:-/home/scu/hostname}
+      START_AS_MODE_CPU: ${SIDECAR_FORCE_CPU_NODE:-0}
+
+    networks:
+      - computational_services_subnet
+
+  dask-sidecar:
+    image: ${DOCKER_REGISTRY:-itisfoundation}/dask-sidecar:${DOCKER_IMAGE_TAG:-latest}
+    init: true
+    deploy:
+      mode: replicated
+      replicas: 2
+      endpoint_mode: dnsrr
+      resources:
+        reservations:
+          cpus: "0.10"
+          memory: "100M"
+    volumes: *sidecar-volumes
     environment:
-      - RABBIT_HOST=${RABBIT_HOST}
-      - RABBIT_PORT=${RABBIT_PORT}
-      - RABBIT_USER=${RABBIT_USER}
-      - RABBIT_PASSWORD=${RABBIT_PASSWORD}
-      - RABBIT_CHANNELS=${RABBIT_CHANNELS}
-      - REDIS_HOST=${REDIS_HOST}
-      - REDIS_PORT=${REDIS_PORT}
-      - POSTGRES_ENDPOINT=${POSTGRES_ENDPOINT}
-      - POSTGRES_USER=${POSTGRES_USER}
-      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
-      - POSTGRES_DB=${POSTGRES_DB}
-      - POSTGRES_HOST=${POSTGRES_HOST}
-      - POSTGRES_PORT=${POSTGRES_PORT}
-      - S3_ENDPOINT=${S3_ENDPOINT}
-      - S3_ACCESS_KEY=${S3_ACCESS_KEY}
-      - S3_SECRET_KEY=${S3_SECRET_KEY}
-      - S3_BUCKET_NAME=${S3_BUCKET_NAME}
-      - STORAGE_ENDPOINT=${STORAGE_ENDPOINT}
-      - REGISTRY_URL=${REGISTRY_URL}
-      - REGISTRY_PATH=${REGISTRY_PATH}
-      - REGISTRY_USER=${REGISTRY_USER}
-      - REGISTRY_PW=${REGISTRY_PW}
-      - SWARM_STACK_NAME=${SWARM_STACK_NAME:-simcore}
-      - SIDECAR_LOGLEVEL=${LOG_LEVEL:-WARNING}
-      - SIDECAR_HOST_HOSTNAME_PATH=${SIDECAR_HOST_HOSTNAME_PATH:-/home/scu/hostname}
-      - START_AS_MODE_CPU=${SIDECAR_FORCE_CPU_NODE:-0}
+      <<: *sidecar-environment
+      DASK_SCHEDULER_ADDRESS: "tcp://dask-scheduler:8786"
 
     networks:
       - computational_services_subnet
 
+  dask-scheduler:
+    image: ${DOCKER_REGISTRY:-itisfoundation}/dask-sidecar:${DOCKER_IMAGE_TAG:-latest}
+    init: true
+    hostname: dask-scheduler
+    healthcheck:
+      test: ["CMD", "curl", "-Lf", "http://dask-scheduler:8787/health"]
```

Comment:
hmm so you put the healthcheck here. and what about the dask-sidecar? actually, you could also use it directly inside the container. since the healtcheck is defined as being inside. Using the local IP would work I think.

Reply:
Actually, you are right, I have just seen that both have ``/health`` endpoint

https://distributed.dask.org/en/latest/http_services.html#http-endpoints

Reply:
I had to enforce workers to start their http server in 8787 port, otherwise it is generated randomly. 
Nonetheless, it is not an issue since they are inside of the container.